met-web/src/components/engagement/view/ActionContext.tsx

```diff
@@ -72,7 +77,36 @@ export const ActionProvider = ({ children }: { children: JSX.Element | JSX.Eleme
                 );
             }
         };
+
+        const fetchWidgets = async () => {
+            try {
+                const result = await getWidgets(Number(engagementId));
+                const whoIsListeningWidget = result.find((w) => w.widget_type_id === WidgetType.WhoIsListening);
+                if (!whoIsListeningWidget) return;
```

Comment:
All good for now. But when we have several widget types this method will become bloated. Perhaps we should remain general at this point like just get widgets (no contacts are any other data). We can create a generic widget card which checks for the type and renders the corresponding widget. If type is who is listening render who is listening widget (switch case maybe). Then that who is listening widget fetches its data.

Reply:
There's also the issue of sorting. At some point we'll have a custom sorting for the widgets which could be easier if we just call get widgets sorted then render them agnostically before each widget calls its data.